src/java.base/share/classes/java/lang/foreign/Arena.java

```diff
@@ -208,6 +208,11 @@
  * @implSpec
  * Implementations of this interface are thread-safe.
  *
+ * @implSpec
+ * Arenas obtained from the factory methods {@linkplain #ofAuto()}, {@linkplain #global()}
+ * {@linkplain #ofConfined()} and, {@linkplain #ofShared()} will return segments
+ * that are zeroed out when invoking {@linkplain Arena#allocate(long, long) allocate()}.
+ *
```

Comment:
I'd expect to find this in the javadoc of each factory method (not in an implSpec). The note says something about those particular methods, not about the Arena type as a whole.

I also think "zero intialized" sounds a bit more professional than "zeroed out". I suggest the following phrasing: 

    Memory segments {@linkPlain #allocate(long, long) allocated} by the returned arena are zero initialized